chevah/python-package
# [#2797] Automate Python distribution on Windows.

Problem
=======

We need an automated way to create a python distribution on windows.

At this stage we don't compile the packages from source but just try to document the way the package is created and to be easy to reproduce.

Changes
=======

The current chevah_build script is designed to run code from source so I have messed it up.

All files are either stored on this repo or on our private PyPi server. In this way we can be sure that the script will work in the future.

When we get files from the net there is the risk of those file to disappear in the future and break the build.

As I drive by I have updated the code which is testing the distribution.
@dumol there is point in different exit code as on multiple failures the exit_code is overwritten.

@alibotean I went for sqlite3 module provided with python but we will need to see if it is the same thing as pysqlite2

How to test
=========

reviewers: @alibotean @dumol 

Manually run the steps used by buildbot

```
./chevah_build build
./chevah_build test
```

check the archive from dist and try to use it for a project.

run the build on buildbot using
```
paver test_remote win-xp
```

The diff is long bug ignore the get-pip.py and ez_setup.py files as they are from upstream.
